core/src/main/java/org/elasticsearch/rest/action/update/RestUpdateAction.java

```diff
@@ -97,6 +101,18 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
             }
         }
 
-        client.update(updateRequest, new RestStatusToXContentListener<>(channel));
+        client.update(updateRequest, new RestBuilderListener<UpdateResponse>(channel) {
+            @Override
+            public RestResponse buildResponse(UpdateResponse response, XContentBuilder builder) throws Exception {
+                builder.startObject();
+                response.toXContent(builder, channel.request());
+                builder.endObject();
+                RestResponse restResponse = new BytesRestResponse(response.status(), builder);
+                if (response.isCreated()) {
+                    restResponse.addHeader("Location", response.getLocation(updateRequest.routing()));
+                }
+                return restResponse;
```

Comment:
Yeah. I think so. I think I'll do a second pass and convert a few other APIs to returning this. I'd planned to pull it out then but I'll do it now.